implementations/python/provekit-lift-py-pytest-witness/src/provekit_pytest_witness/witness.py

```
@@ -79,8 +79,80 @@ class Witness:
+    fd, out_path = tempfile.mkstemp(suffix=".json", prefix="pvk-witness-")
+    os.close(fd)
+    try:
+        # Load the capture plugin STANDALONE (bare module name), with its own
+        # directory on PYTHONPATH. `_witness_collect` imports only json/os, so it
+        # loads without pulling in the rest of the package (which would need the
+        # whole kit dependency chain on the path). This makes the per-test run
+        # robust to cwd and to a partial PYTHONPATH.
+        plugin_dir = os.path.dirname(os.path.abspath(__file__))
+        env_pp = os.pathsep.join(p for p in (plugin_dir, os.environ.get("PYTHONPATH", "")) if p)
+        env = dict(os.environ, PROVEKIT_WITNESS_OUT=out_path, PYTHONPATH=env_pp)
+        subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)